react/features/authentication/components/native/AuthenticationPage.tsx

```diff
@@ -0,0 +1,85 @@
+import { Route } from '@react-navigation/native';
+import React, { PureComponent } from 'react';
+import { WithTranslation } from 'react-i18next';
+import { View, ViewStyle } from 'react-native';
+import { WebView } from 'react-native-webview';
+import { connect } from 'react-redux';
+
+import { IStore } from '../../../app/types';
+import { translate } from '../../../base/i18n/functions';
+import JitsiScreen from '../../../base/modal/components/JitsiScreen';
+import LoadingIndicator from '../../../base/react/components/native/LoadingIndicator';
+
+import styles, { INDICATOR_COLOR } from './styles';
+
+
+interface IProps extends WithTranslation {
+
+    dispatch: IStore['dispatch'];
+
+    /**
+     * Default prop for navigating between screen components(React Navigation).
+     */
+    navigation: any;
+
+    /**
+     * Default prop for navigating between screen components(React Navigation).
+     */
+    route: Route<'', { tokenAuthServiceUrl: string; }>;
+}
+
+/**
+ * Implements a React native component that displays the authentication page for a specific room.
+ */
+class AuthenticationPage extends PureComponent<IProps> {
+
+    /**
+     * Initializes a new instance.
+     *
+     * @inheritdoc
+     */
+    constructor(props: IProps) {
+        super(props);
+
+        this._renderLoading = this._renderLoading.bind(this);
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const { route } = this.props;
+        const tokenAuthServiceUrl = route.params?.tokenAuthServiceUrl;
+
+        return (
+            <JitsiScreen
+                style = { styles.backDrop }>
+                <WebView
```

Comment:
Here we need to add some extra behavior. The WebView should redirect to a URL with a token. We need to intercept that redirect and navigate there ourselves.

This is because while the redirect might be caught as a deep-link for our domains, it won't for others.

In addition it makes the mechanism more generic.